src/widget/wlabel.cpp

```diff
@@ -43,26 +45,62 @@ void WLabel::setup(QDomNode node, const SkinContext& context) {
     setPalette(pal);
 
     // Text
-    if (context.hasNode(node, "Text"))
-        m_qsText = context.selectString(node, "Text");
-    setText(m_qsText);
+    if (context.hasNodeSelectString(node, "Text", &m_skinText)) {
+        setText(m_skinText);
+    }
 
     // Font size
-    if (context.hasNode(node, "FontSize")) {
-        int fontsize = context.selectString(node, "FontSize").toInt();
+    QString strFontSize;
+    if (context.hasNodeSelectString(node, "FontSize", &strFontSize)) {
+        int fontsize = strFontSize.toInt();
         setFont(QFont("Helvetica", fontsize, QFont::Normal));
     }
 
     // Alignment
-    if (context.hasNode(node, "Alignment")) {
-        if (context.selectString(node, "Alignment").toLower() == "right") {
+    QString alignment;
+    if (context.hasNodeSelectString(node, "Alignment", &alignment)) {
+    	alignment = alignment.toLower();
+        if (alignment == "right") {
             setAlignment(Qt::AlignRight | Qt::AlignVCenter);
-        } else if (context.selectString(node, "Alignment").toLower() == "center") {
+        } else if (alignment == "center") {
             setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
-        } else if (context.selectString(node, "Alignment").toLower() == "left") {
+        } else if (alignment == "left") {
             setAlignment(Qt::AlignLeft | Qt::AlignVCenter);
         }
     }
+
+    // Adds an ellipsis to turncated text
+    QString elide;
+    if (context.hasNodeSelectString(node, "Elide", &elide)) {
+    	elide = elide.toLower();
+        if (elide == "right") {
+            m_elideMode = Qt::ElideRight;
+        } else if (elide == "middle") {
+            m_elideMode = Qt::ElideMiddle;
+        } else if (elide == "left") {
+            m_elideMode = Qt::ElideLeft;
+        } else if (elide == "none") {
+            m_elideMode = Qt::ElideNone;
+        }
```

Comment:
one thing I want to start doing is display a warning when a user picks an invalid value.   so like else { skinwarning("invalid elide value: %s")}.  same above for alignment.  TODO is ok